packages/ListBox/src/components/WithTags/WithTags.js

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import ListBox from "../..";
+
+import Pill, { Pills } from "../Pill";
+import { filter } from "../../helpers/filter";
+import * as sc from "./WithTags.styles";
+
+const propTypes = {
+  /** Expect <ListBoxWithTags.Option /> */
+  children: PropTypes.node.isRequired,
+  /** filter function for the ListBoxWithTags can be pair with ListBoxWithTags.filter  */
+  filter: PropTypes.func,
+  /** String message to be display when there are not results  */
+  noResultsMessage: PropTypes.node,
+  /** Callback whenever the user change a selection on the ListBoxWithTags  */
+  onChange: PropTypes.func,
+  /** Callback whenever the user input a new custom option like some@email.com, pass undefined to ignore this behaviour */
+  onCustomOption: PropTypes.func,
+  /** Callback once a pill is remove from the Trigger */
+  onRemove: PropTypes.func,
+  /** Regex that match the input of the user and reports to onCustomOption. The default is a basic email regex */
+  regexCustomOption: PropTypes.instanceOf(RegExp),
+  /** Render prop to override the default Pill style, see example for it's uses.  */
+  renderPill: PropTypes.func,
+  /** An array of id that helps the ListBoxWithTags to known what elements are selected  */
+  selectedOptions: PropTypes.arrayOf(PropTypes.shape({})),
+};
+
+const defaultProps = {
+  filter: undefined,
+  noResultsMessage: null,
+  onChange: () => {},
+  onCustomOption: null,
+  onRemove: () => {},
+  regexCustomOption: /^.+@.+\..+$/,
+  renderPill: null,
+  selectedOptions: null,
+};
+
+const renderTrigger = ({ t, size, selectedOptions, onRemove, renderPill }) => (...args) => {
+  const [, , , attributes] = args;
+  const { propsForTrigger, refTrigger, dispatch, types, handleKeyDown, handleKeyUp } = attributes;
+
+  function handleClick(event) {
+    event.stopPropagation();
+    if (event.target.dataset.pkaAnchor === "listbox-tags-pill-delete") {
+      return;
+    }
+
+    if (event.key === "Enter" || event.key === " ") return;
+    dispatch({ type: types.togglePopover });
+  }
+
+  const handleRemove = option => event => {
+    event.stopPropagation();
+    onRemove(option);
+  };
+
+  return (
+    <sc.Trigger
+      ref={refTrigger}
+      {...propsForTrigger()}
+      onClick={handleClick}
+      onKeyUp={handleKeyUp}
+      onKeyDown={handleKeyDown}
+      size={size}
+    >
+      <Pills>
+        {selectedOptions.map(item => {
+          if (typeof renderPill === "function") {
+            return renderPill({ option: item, Pill, onRemove: handleRemove(item) });
```

Comment:
Any benefit to passing `Pill` in parameters vs exporting it from the package? Technically there's no need to pass it in since it's the same every time

Reply:
Mmm, for this pass it's easier to expose it like this, I will extract the Pill to its own package soon, and then can be imported, don't want to export it as part of the ListBox package.

Meanwhile, this allowed us to expose it easily and also can be fixed quickly whenever we do a breaking change or we can even just import the `Pill` inside of ListBox and still be backward compatible.